### PR TITLE
PyCon Open Spaces example content

### PR DESCRIPTION
**Pull request by kattni**
Please test as many of these as possible. @makermelissa I was unable to add you as a reviewer, so I am adding you here.

While testing, please verify the following:
* Each example works.
* Each example has a docstring with a quick explanation
* Examples requiring external hardware have connection instructions in the docstring
* Examples that require an external library have a huge note in the docstring stating the need and indicating the library or libraries needed.
* Examples are clear and simple as possible (which in some cases means longer for readability)

Hardware needed to test all the examples:
* 2 Circuit Playground Expresses (1 will work for everything but the IR examples)
* A micro servo (alligator-to-jumper-wire clips are easiest to connect it with)
* A 10k potentiometer (and alligator clips)
* A 30-pixel NeoPixel strip with alligator clips

Please leave do not merge until everything has been tested.

Thank so much!

### CODE REVIEW

**Comment by makermelissa** [comment]
I only have 1 CPX. Should I order another or just not test IR? I should have enough of the other stuff to kludge together some test accessories.

**Comment by kattni** [comment]
@makermelissa Don't worry about testing IR. Most of the examples do not require any hardware at all, so even if you don't have the listed hardware, you can still do quite a bit.

**Comment by makermelissa** [comment]
Ok, sounds good.

**Comment by makermelissa** [comment]
Parts ordered, but I'm going to start testing the ones that don't require parts until then. This may take a bit. :)

**Comment by makermelissa** [comment]
Which version of CircuitPython should we be testing these on? AFAIK, they shipped with CP 3.1.2, but CP 4 should be out by then. For now I'm going to test with CP 4.

**Comment by dhalbert** [comment]
~~They shipped with 3.1.2, so test with that. 4.0.0 will probably be final by PyCon, but we couldn't be sure, and there will always be bugs. 3.1.2 has a longer history of general use. If you want to test with both, that would be great, but not necessary.~~ Edit: I thought I heard that at a meeting, but I was wrong. Sorry!

**Comment by makermelissa** [comment]
Thanks @danh.

**Comment by kattni** [comment]
@dhalbert @makermelissa That is incorrect, they shipped with 4.0.0-beta.7. Please test with that version. Thanks!

**Comment by makermelissa** [comment]
Ok, thanks @kattni

**Comment by siddacious** [comment]
@makermelissa Since I have most of the parts required, I'll start on the ones that do require them and then we'll pass somewhere in the middle? I should be able to gets started later today.

**Comment by makermelissa** [comment]
Sounds good. It also doesn’t hurt if some stuff is tested by both of us.

**Comment by kattni** [comment]
Hi all! @brentru may be joining in the testing. So, please continue to communicate about where you're at with testing so everyone is on the same page. Thanks!!

**Comment by siddacious** [comment]
Right, that's what I meant by "pass in the middle" assuming we were both going to test most things.

**Comment by makermelissa** [comment]
Ah right. Sounds good.

**Comment by makermelissa** [comment]
Just one minor thing I found. In `cpx_buttons_1_neopixel.py` it doesn't quite do what the description says. For instance, if you hold button B and then press button A at the same time and then let go of button B, both lights remain on while you continue holding button A.

**Comment by makermelissa** [comment]
Same thing with `cpx_buttons_neopixels.py` and `cpx_start_stop_tone.py`

**Comment by makermelissa** [comment]
I've tested everything in Circuit_Playground_Express_Examples and everything in Circuit_Playground_Express_cpx_Library_Examples (except IR examples for now). I'll finish testing the rest of the examples tomorrow when I get the additional parts.

File: Accessory_Examples/buttons_servo.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import board
+import pulseio
+from adafruit_motor import servo
+from adafruit_circuitplayground.express import cpx
+
+pwm = pulseio.PWMOut(board.A1, duty_cycle=2 ** 15, frequency=50)
+servo = servo.Servo(pwm)
+
+while True:
+    if cpx.button_A:
```
Comment (makermelissa) [inline Accessory_Examples/buttons_servo.py]: I needed to change this to lowercase button_a to get it to function.

File: Accessory_Examples/buttons_servo.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        servo.angle = 180
+    elif cpx.button_B:
```
Comment (makermelissa) [inline Accessory_Examples/buttons_servo.py]: Same with the button_b

**Review by makermelissa — CHANGES_REQUESTED** [review]
I finished testing all examples. I had some earlier comments and there was only one example that wasn't working which I requested changes on. Everything else is looking good.

**Comment by ntoll** [comment]
You folks rock.

**Comment by kattni** [comment]
@ntoll Thanks! 😊 And thank you for the pulsing Python logo example - I've included it in this PR to be part of our Open Spaces content!

**Comment by kattni** [comment]
@makermelissa @siddacious I would like to merge this PR tonight sometime, as I need to continue adding to this repo. If anyone has any further suggestions, please get them in. We can always do another PR if any issues are found after merging.

File: Accessory_Examples/buttons_servo.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+"""Connect the blue clip (yellow wire) on the servo to A1.
+Connect the black clip (brown wire) on the servo to a GND pad.
+Connect the red clip (red wire) on the servo to a 3.3v pad.
```
Comment (siddacious) [inline Accessory_Examples/buttons_servo.py]: These should read "Connect the x clip to the y wire on the servo to z pad"

Comment (siddacious) [reply Accessory_Examples/buttons_servo.py]: Also the yellow wire looks kinda orange

File: Accessory_Examples/potentiometer_neopixel_strip.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
+"""Connect the blue clip on the potentiometer to A0, and the white clip on the NeoPixel strip to A7.
```
Comment (siddacious) [inline Accessory_Examples/potentiometer_neopixel_strip.py]: I'm now kinda thinking that these should also list the required hardware, then the connections. Should be obvious but it isn't always.

Comment (siddacious) [reply Accessory_Examples/potentiometer_neopixel_strip.py]: I think these are meant to be followed with provided hardware that matches the colors given but it's somewhat unclear if that's not the case.

**Review by makermelissa — APPROVED** [review]
Looks good. Thanks.

File: Circuit_Playground_Express_Examples/light_sensor.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+shining a flashlight on your CPX, or covering the light sensor with your finger to see the values
+increase and decrease."""
+import time
+import board
+import analogio
+
+light_sensor = analogio.AnalogIn(board.LIGHT)
+
+while True:
+    print(light_sensor.value)
```
Comment (siddacious) [inline Circuit_Playground_Express_Examples/light_sensor.py:11]: It is a little confusing that this prints different values than the similar example using the cpx lib

File: Circuit_Playground_Express_cpx_Library_Examples/cpx_light_plotter.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+"""If you're using Mu, this example will plot the light levels from the light sensor (located next
+to the eye) on your CPX. Try shining a flashlight on your CPX, or covering the light sensor to see
+the plot increase and decrease."""
+import time
+from adafruit_circuitplayground.express import cpx
+
+while True:
+    print("Light:", cpx.light)
+    print((cpx.light,))
+    time.sleep(1)
```
Comment (siddacious) [inline Circuit_Playground_Express_cpx_Library_Examples/cpx_light_plotter.py]: Personally I think the graph/plot is more interactive with a shorter sleep

File: Circuit_Playground_Express_cpx_Library_Examples/cpx_shake.py [hunk 7]
```diff
@@ -0,0 +1,6 @@
+"""This example prints to the serial console when you shake the board!"""
+from adafruit_circuitplayground.express import cpx
+
+while True:
+    if cpx.shake():
+        print("Shake detected!")
```
Comment (siddacious) [inline Circuit_Playground_Express_cpx_Library_Examples/cpx_shake.py:9]: Either I have a literal limp wrist or the threshold on shake is too high. Or both. I had to shake what seemed excessively hard to get a shake to register

File: Circuit_Playground_Express_cpx_Library_Examples/cpx_slide_switch_red_led.py [hunk 8]
```diff
@@ -0,0 +1,11 @@
+"""This example uses the slide switch to control the little red LED."""
+from adafruit_circuitplayground.express import cpx
+
+while True:
+    if cpx.switch:
+        cpx.red_led = True
+    else:
+        cpx.red_led = False
+
+# Can also be written as:
```
Comment (siddacious) [inline Circuit_Playground_Express_cpx_Library_Examples/cpx_slide_switch_red_led.py]: This isn't needed if there is going to be a separate example that does the same thing

File: Circuit_Playground_Express_cpx_Library_Examples/cpx_temperature_neopixels.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+less sensitive to temperature changes.
+"""
+import time
+from adafruit_circuitplayground.express import cpx
+import simpleio
+
+cpx.pixels.auto_write = False
+cpx.pixels.brightness = 0.3
+
+# Set these based on your ambient temperature for best results!
```
Comment (siddacious) [inline Circuit_Playground_Express_cpx_Library_Examples/cpx_temperature_neopixels.py]: This should probably mention that it's in celcius

File: Circuit_Playground_Express_cpx_Library_Examples/cpx_temperture_plotter.py [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+"Plotter" to open it, and place your finger over the sensor to see the numbers change. The
+sensor is located next to the picture of the thermometer on the CPX."""
+import time
+from adafruit_circuitplayground.express import cpx
+
+while True:
+    print("Temperature C:", cpx.temperature)
+    print("Temperature F:", cpx.temperature * 1.8 + 32)
+    print((cpx.temperature, cpx.temperature * 1.8 + 32))
+    time.sleep(1)
```
Comment (siddacious) [inline Circuit_Playground_Express_cpx_Library_Examples/cpx_temperture_plotter.py]: as above, I think a smaller sleep will make the plot more interactive and interesting

File: Circuit_Playground_Express_cpx_Library_Examples/cpx_touch_tone_machine.py [hunk 11]
```diff
@@ -0,0 +1,29 @@
+"""This example requires seven wav files from the Additional_Content folder on the PyCon2019 GitHub
+repo. Download the files and copy them to your CIRCUITPY drive. Once copied, try touching the touch
+pads on A1-A7 to hear a funky scale of notes!"""
```
Comment (siddacious) [inline Circuit_Playground_Express_cpx_Library_Examples/cpx_touch_tone_machine.py:3]: The "I had to memorize all the major and minor scales" part of my brain hates this but it works great!

**Review by siddacious — CHANGES_REQUESTED** [review]
Everything works fine but a few doc strings were a bit confusing, possibly due to me not using what I assume is going to be hardware that is provided. A handful of opinions about possible other tweaks but take them as you will.

**Review by siddacious — APPROVED** [review]

**Comment by siddacious** [comment]
great, thanks!

**Pull request merged**